ui/src/components/Nodes/NodeHierarchyPanel.vue

```diff
@@ -0,0 +1,200 @@
+<template>
+  <div class="button-panel">
+      <FeatherButton primary class="category-btn" @click="onClearAll">Clear All</FeatherButton>
+  </div>
+  <FeatherExpansionPanel>
+    <template v-slot:title>
+      <div v-if="selectedCategoryCount">
+        <span>{{ `Categories (${selectedCategoryCount})` }}</span>
+        <FeatherButton icon="Clear" @click="onClearCategories">
+          <FeatherIcon :icon="clearIcon"> </FeatherIcon>
+        </FeatherButton>
+      </div>
+      <div v-else>Categories</div>
+    </template>
+    <FeatherList class="category-list">
+      <FeatherListItem
+        v-for="cat of categories"
+        :selected="isCategorySelected(cat)"
+        :key="cat.name"
+        @click="onCategoryClick(cat)">
+        {{ cat.name }}
+      </FeatherListItem>
+    </FeatherList>
+  </FeatherExpansionPanel>
+  <FeatherExpansionPanel>
+    <template v-slot:title>
+      <div v-if="selectedFlowCount">
+        <span>{{ `Flows (${selectedFlowCount})` }}</span>
+        <FeatherButton icon="Clear" @click="onClearFlows">
+          <FeatherIcon :icon="clearIcon"> </FeatherIcon>
+        </FeatherButton>
+      </div>
+      <div v-else>Flows</div>
+    </template>
+    <FeatherList class="category-list">
+      <FeatherListItem
+        v-for="flow of flowTypes"
+        :selected="isFlowSelected(flow)"
+        :key="flow"
+        @click="onFlowClick(flow)">
+        {{ flow }}
+      </FeatherListItem>
+    </FeatherList>
+  </FeatherExpansionPanel>
+  <FeatherExpansionPanel>
+    <template v-slot:title>
+      <div v-if="selectedLocationCount">
+        <span>{{ `Locations (${selectedLocationCount})` }}</span>
+        <FeatherButton icon="Clear" @click="onClearLocations">
+          <FeatherIcon :icon="clearIcon"> </FeatherIcon>
+        </FeatherButton>
+      </div>
+      <div v-else>Locations</div>
+    </template>
+    <FeatherList class="category-list">
+      <FeatherListItem
+        v-for="loc of locations"
+        :selected="isLocationSelected(loc)"
+        :key="loc.name"
+        @click="onLocationClick(loc)">
+        {{ loc.name }}
+      </FeatherListItem>
+    </FeatherList>
+  </FeatherExpansionPanel>
+</template>
+
+<script setup lang="ts">
+import ClearIcon from '@featherds/icon/action/Cancel'
+import { FeatherButton } from '@featherds/button'
+import { FeatherExpansionPanel } from '@featherds/expansion'
+import { FeatherIcon } from '@featherds/icon'
+import { FeatherList, FeatherListItem } from '@featherds/list'
+import { useStore } from 'vuex'
+import { Category, MonitoringLocation } from '@/types'
+
+const store = useStore()
+const clearIcon = ref(ClearIcon)
+const categories = computed<Category[]>(() => store.state.hierarchyModule.categories)
```

Comment:
I believe some of these computed variables are only being used in the template for iteration and displaying, just wanted to let you know that you can use store variables directly in the template with full reactive functionality. If you're doing this though for readability (not seeing store.state.hierarchyModule littered everywhere) then feel free to ignore me! :)